pydoctor/templatewriter/__init__.py

```diff
@@ -56,130 +73,162 @@ def writeSummaryPages(self, system: System) -> None:
         """
-        Create a concrete template object.
-        Type depends on the file extension.
-
-        Warns if the template cannot be created.
+        Scan a directory for templates. 
 
-        @param path: A L{Path} that should point to a HTML, CSS or JS file.
-        @returns: The template object or C{None} if file is invalid.
+        @param path: A L{Path} or L{Traversable} object that should point to a template folder.
+        @param subdir: The subdirectory inside the template folder structure that holds this template directory. 
+        @raises FailedToCreateTemplate: If the path is not a directory or do not exist. 
         """
-        if path.suffix.lower() in cls.TEMPLATE_FILES_SUFFIX:
-            try:
-                with path.open('r', encoding='utf-8') as fobj:
-                    text = fobj.read()
-            except IOError as e:
-                warnings.warn(f"Cannot load Template: {path.as_posix()}. I/O error: {e}")
+        if not path.is_dir():
+            raise FailedToCreateTemplate(f"Template folder do not exist or is not a directory: {path}")
+        subdir = subdir.joinpath(path.name) if subdir else PurePath()
+        for entry in path.iterdir():
+            if entry.is_dir():
+                yield from Template.fromdir(entry, subdir=subdir)
             else:
-                if path.suffix.lower() == '.html':
-                    return _HtmlTemplate(name=path.name, text=text)
-                else:
-                    return _StaticTemplate(name=path.name, text=text)
-        else:
-            warnings.warn(f"Cannot create Template: {path.as_posix()} is not recognized as template file. "
-                f"Template files must have one of the following extensions: {', '.join(cls.TEMPLATE_FILES_SUFFIX)}")
-        return None
-
-    def is_empty(self) -> bool:
-        """
-        Does this template contain nothing except whitespace?
-        Empty templates will not be rendered.
-        """
-        return len(self.text.strip()) == 0
+                template = Template.fromfile(entry, subdir=subdir)
+                if template:
+                    yield template
 
-    @abc.abstractproperty
-    def version(self) -> int:
+    @classmethod
+    def fromfile(cls, path: Union[Traversable, Path], subdir: Optional[PurePath] = None) -> Optional['Template']:
         """
-        Template version, C{-1} if no version.
-
-        HTML Templates should have a version identifier as follow::
-
-            <meta name="pydoctor-template-version" content="1" />
+        Create a concrete template object.
+        Type depends on the file extension.
 
-        This is always C{-1} for CSS and JS templates.
+        @param path: A L{Path} or L{Traversable} object that should point to a template file.
+        @param subdir: The subdirectory inside the template folder structure that holds this template file. 
```

Comment:
First, the `subdir` parameter is used only after the first recursion in the `Template.fromdir` call. 

I can try to refactor this code to look like what you suggested but if it's not a requirement for this PR, I'll pass since that current implementation is working good and is well tested.